lib/AST/Module.cpp

```diff
@@ -3649,6 +3649,33 @@ ASTScope &SourceFile::getScope() {
   return *Scope.get();
 }
 
+bool SourceFile::matchesFileID(StringRef fileID) const {
+  // Never match with SourceFiles that do not correpond to a file on disk
+  if (getFilename().empty())
+    return false;
+
+  auto parsed = parseFileID(fileID);
+  if (!parsed)
+    return false;
+
+  auto moduleName = parsed->first;
+  auto fileName = parsed->second;
+  return moduleName == getParentModule()->getNameStr() &&
+         fileName == llvm::sys::path::filename(getFilename());
+}
+
+std::optional<std::pair<StringRef, StringRef>>
+SourceFile::parseFileID(StringRef fileID) {
+  auto names = fileID.split('/');
```

Comment:
Would we always have a single `/`?

Reply:
Yes, I think so. The format of a fileID is always `modulename/filename`.

Reply:
Added a comment to mention this